acceptancetests/clean_resources.py

```
@@ -27,28 +27,62 @@ def get_regions(args, env):
-            except:
-                logging.info('Unable to delete non-instance groups {}'.format(non_instance_groups.keys()))
-            for group_id in unclean:
-                logging.debug('Cannot delete {}'.format(all_groups[group_id]))
-            for group_id in unclean:
-                non_instance_groups.pop(group_id, None)
-            try:
-                substrate.destroy_security_groups(non_instance_groups.values())
-            except:
-                logging.debug('Failed to delete groups {}'.format(non_instance_groups.values()))
+            if substrate is not None:
+                logging.info('Cleaning resources in {}.'
+                             .format(substrate.region))
+                all_groups = dict(substrate.iter_security_groups())
+                instance_groups = dict(substrate.iter_instance_security_groups())
+                logging.info('{} items found from instance groups on {} \n{}'
+                             .format(str(len(instance_groups)),
+                                     region,
+                                     '\n'.join(instance_groups)))
+                non_instance_groups = dict((k, v) for k, v in all_groups.items()
+                                           if k not in instance_groups)
+                logging.info('{} items found from non-instance groups on {} \n{}'
+                             .format(str(len(non_instance_groups)),
+                                     region,
+                                     '\n'.join(non_instance_groups)))
+                try:
+                    logging.info('{} detached interfaces will be deleted from {} \n{}'
+                                 .format(str(len(non_instance_groups.keys())),
+                                         region,
+                                         '\n'.join(non_instance_groups.keys())))
+                    unclean = substrate.delete_detached_interfaces(
+                        non_instance_groups.keys())
```

If there is a instance group in the list that causes the exception to be raised we loose any information for any unclean or any follow up instances.

Perhaps a better mechanism would be to loop through each instance and call delete_detached_interfaces on it (accumulating any unclean).

This way we can log but ignore any exceptions raised by the call, ensuring that we get through the list of instances and try _all_ instances. Which means that any exception raised by delete_detached_interfaces won't leave us with an undefined unclean list.